examples/codegen/codegen.cpp

```diff
@@ -79,7 +97,8 @@ int main() {
     ur_check(urInit(UR_DEVICE_INIT_FLAG_GPU, loader_config));
 
     auto adapters = get_adapters();
-    auto platforms = get_platforms(adapters);
+    auto supported_adapters = get_supported_adapters(adapters);
+    auto platforms = get_platforms(supported_adapters);
     auto gpus = get_gpus(platforms.front());
```

Comment:
You need to check if the platforms aren't empty. [Calling `front()` on an empty container is UB](https://en.cppreference.com/w/cpp/container/vector/front).

Reply:
If there are no platforms `get_platforms()` will `throw std::runtime_error("No platforms available.");` so I think its okay?

Reply:
right, I haven't noticed that.